sql/views.sql

```diff
@@ -259,5 +259,121 @@ CREATE OR REPLACE VIEW timescaledb_information.data_node AS
+        WHEN ( dim.column_type = 'TIMESTAMP'::regtype OR
+               dim.column_type = 'TIMESTAMPTZ'::regtype OR
+               dim.column_type = 'DATE'::regtype )
+        THEN NULL
+        ELSE dimsl.range_end
+    END as integer_range_end,
+    CASE WHEN srcch.compressed_chunk_id is not null THEN 'true'
+         ELSE 'false'
+    END as is_compressed,
+    pgtab.spcname as chunk_table_space,
+    chdn.node_list
+FROM _timescaledb_catalog.chunk srcch 
+      INNER JOIN _timescaledb_catalog.hypertable ht ON ht.id = srcch.hypertable_id
+      INNER JOIN _timescaledb_catalog.chunk_constraint chcons ON srcch.id = chcons.chunk_id
+      INNER JOIN _timescaledb_catalog.dimension dim ON srcch.hypertable_id = dim.hypertable_id
+      INNER JOIN _timescaledb_catalog.dimension_slice dimsl ON dim.id = dimsl.dimension_id
+                                and chcons.dimension_slice_id = dimsl.id
+      INNER JOIN
+       ( SELECT relname, reltablespace, nspname as schema_name 
+                FROM pg_class , pg_namespace WHERE
+                pg_class.relnamespace = pg_namespace.oid) cl 
+       ON srcch.table_name = cl.relname and srcch.schema_name = cl.schema_name
+       LEFT OUTER JOIN pg_tablespace pgtab ON pgtab.oid = reltablespace
+       left outer join (
+               SELECT chunk_id, array_agg(node_name ORDER BY node_name) as node_list
+                            FROM _timescaledb_catalog.chunk_data_node
+                            GROUP BY chunk_id) chdn
+       ON srcch.id = chdn.chunk_id 
+ WHERE srcch.dropped is false 
+      and ht.compressed = false ) finalq
+WHERE chunk_dimension_num = 1;
+ ;
+
+CREATE OR REPLACE VIEW timescaledb_information.dimensions
+AS
+SELECT 
+    ht.schema_name as hypertable_schema,
+    ht.table_name as hypertable_name,
+    rank() over(partition by hypertable_id order by dim.id) as dimension_number,
+    dim.column_name,
+    dim.column_type,
+    CASE WHEN dim.interval_length is NULL 
+         THEN 'Space' 
+         ELSE 'Time' 
+    END as dimension_type,
+    CASE WHEN dim.interval_length is NOT NULL THEN
+              CASE 
+              WHEN dim.column_type = 'TIMESTAMP'::regtype OR 
+                   dim.column_type = 'TIMESTAMPTZ'::regtype OR 
+                   dim.column_type = 'DATE'::regtype 
+              THEN _timescaledb_internal.to_interval(dim.interval_length) 
+              ELSE NULL 
+              END
+    END as time_interval, 
+    CASE WHEN dim.interval_length is NOT NULL THEN
+              CASE 
+              WHEN dim.column_type = 'TIMESTAMP'::regtype OR 
+                   dim.column_type = 'TIMESTAMPTZ'::regtype OR 
+                   dim.column_type = 'DATE'::regtype 
+              THEN  NULL
+              ELSE dim.interval_length 
+              END
+    END as integer_interval, 
+    dim.integer_now_func,
+    dim.num_slices as num_partitions
+FROM _timescaledb_catalog.hypertable ht, _timescaledb_catalog.dimension dim
+WHERE dim.hypertable_id = ht.id
```

Comment:
You are mixing the styles of inner joins. I have no strong opinion either way, but it is good to be consistent.

Reply:
I use INNER JOIN syntax only where I am forced to : i.e. presence of LEFT OUTER JOINs etc.